src/Control/Effect/Sum.hs

```diff
@@ -20,54 +18,25 @@ instance (HFunctor f, HFunctor g) => HFunctor (f :+: g)
 instance (Effect f, Effect g)     => Effect   (f :+: g)
 
 
-type Member sub sup = (Inject sub sup, Project sub sup)
-
-
-class Inject (sub :: (* -> *) -> (* -> *)) sup where
+class Member (sub :: (* -> *) -> (* -> *)) sup where
   inj :: sub m a -> sup m a
 
-instance Inject t t where
+instance Member t t where
   inj = id
 
 instance {-# OVERLAPPABLE #-}
-         Inject t (l1 :+: l2 :+: r)
-      => Inject t ((l1 :+: l2) :+: r) where
+         Member t (l1 :+: l2 :+: r)
+      => Member t ((l1 :+: l2) :+: r) where
   inj = reassoc . inj where
     reassoc (L l)     = L (L l)
     reassoc (R (L l)) = L (R l)
     reassoc (R (R r)) = R r
 
 instance {-# OVERLAPPABLE #-}
-         Inject l (l :+: r) where
+         Member l (l :+: r) where
   inj = L
 
 instance {-# OVERLAPPABLE #-}
-         Inject l r
-      => Inject l (l' :+: r) where
+         Member l r
+      => Member l (l' :+: r) where
   inj = R . inj
-
-
-class Project (sub :: (* -> *) -> (* -> *)) sup where
-  prj :: sup m a -> Maybe (sub m a)
```

Comment:
`InterposeC` represented the only use of `Project` in the library, so I’ve opted to remove it. The justification for this is a little bit weaker, IMO, but in short: it’s impolite, and probably surprising, to (metaphorically) wander into other people’s houses uninvited.

We currently use `prj` in three places in `semantic` (all of which represent some variation on interposition which should be changed), as well as several places in `semantic-core` and `path` (all of which represent syntactic operations which we could support by redefining `Project`/`prj` locally).

The latter category make me wonder if we should retain this for syntax use cases, but tbh `bound`-style syntax having the same kind as higher-order effects is at best a pun, because the syntax is generally a very different _shape_:

- Nullary constructors do not represent abnormal termination of control.
- Constructors are designed to hold terms, rather than continuations per se.

This makes `Carrier` instances beyond the typical `Term` at best inconvenient, and perhaps warrants a `fused-syntax` package enabling e.g. algebras for non-`Monad`ic carriers. (And perhaps that package and `fused-effects` will someday share some higher-order syntax such as sums. Life is full of possibility.)

Thus, while I’m slightly concerned about this, I am currently, on the balance, convinced that removing it is right for the library; but I would nevertheless particularly welcome a gut check.